caafe/caafe.py

```diff
@@ -98,14 +153,17 @@ def build_prompt_from_df(ds, df, iterative=1):
 def generate_features(
     ds,
     df,
-    model="gpt-3.5-turbo",
-    just_print_prompt=False,
-    iterative=1,
-    metric_used=None,
-    iterative_method="logistic",
-    display_method="markdown",
-    n_splits=10,
-    n_repeats=2,
+    model: str = "gpt-3.5-turbo",
+    just_print_prompt: bool = False,
+    iterative: int = 1,
+    metric_used: Optional[str] = None,
+    iterative_method: str = "logistic",
+    display_method: str = "markdown",
+    n_splits: int = 10,
+    n_repeats: int = 2,
+    llm_provider: str = "openai",
+    region_name: str = "us-west-2",
```

Comment:
perhaps better instead of `region_name`, have `llm_provider_kwargs: dict = None`

and make `llm_provider` accept a class as well, so that you can do:

```
if llm_kwargs is None:
    llm_kwargs = {}
if isinstance(llm_provider, str):
    if llm_provider == "openai":
        llm_provider = OpenAIClient
    elif llm_provider == "bedrock":
        llm_provider = BedrockClient
    else:
        raise ValueError(f"Unsupported LLM provider: {llm_provider}")
llm_client = llm_provider(**llm_provider_kwargs)
```